king_phisher/client/application.py

```diff
@@ -552,6 +555,17 @@ def do_config_load(self, load_defaults):
 			for key, value in client_template.items():
 				if not key in self.config:
 					self.config[key] = value
+			env_proxy = os.environ.get('HTTPS_PROXY')
+			if env_proxy is not None:
+				proxy_url = urllib.parse.urlparse(env_proxy)
+				if proxy_url.hostname and not proxy_url.scheme or proxy_url.scheme and not proxy_url.hostname:
+					self.logger.error('invalid proxy url scheme or hostname')
+					return
+				if env_proxy != self.config['proxy.url']:
+					self.logger.warning('proxy environment variable does not match the user configuration')
+					self.config['proxy.url'] = env_proxy
```

Comment:
We would want this set either way. Whether `env_proxy` matches it or not. If it does match it, logging an `info` message would be good while still logging a `warning` message if it does not match like you are doing.